airbyte-integrations/connectors/destination-postgres/src/main/kotlin/io/airbyte/integrations/destination/postgres/check/PostgresCloudChecker.kt

```diff
@@ -0,0 +1,33 @@
+package io.airbyte.integrations.destination.postgres.check
+
+import io.airbyte.cdk.command.AIRBYTE_CLOUD_ENV
+import io.airbyte.cdk.load.check.DestinationCheckerV2
+import io.airbyte.cdk.ssh.SshNoTunnelMethod
+import io.airbyte.integrations.destination.postgres.client.PostgresAirbyteClient
+import io.airbyte.integrations.destination.postgres.spec.PostgresConfiguration
+import io.airbyte.integrations.destination.postgres.spec.SslModeRequire
+import io.airbyte.integrations.destination.postgres.spec.SslModeVerifyCa
+import io.airbyte.integrations.destination.postgres.spec.SslModeVerifyFull
+import io.micronaut.context.annotation.Requires
+import jakarta.inject.Singleton
+
+@Singleton
+@Requires(env = [AIRBYTE_CLOUD_ENV])
+class PostgresCloudChecker(
+    private val postgresConfiguration: PostgresConfiguration,
+    postgresAirbyteClient: PostgresAirbyteClient,
+) : DestinationCheckerV2 {
+
+    private val postgresOssChecker = PostgresOssChecker(postgresAirbyteClient, postgresConfiguration)
+
+    override fun check() {
```

Comment:
this is meant to be doing the same we were doing [here](https://github.com/airbytehq/airbyte/blob/7df8a9430f5ba0b4a944e04f6af87961d27f1323/airbyte-integrations/connectors/destination-postgres-strict-encrypt/src/main/kotlin/io/airbyte/integrations/destination/postgres/PostgresDestinationStrictEncrypt.kt#L29-L51)